test/eventmesh.LockManager.spec.js

```diff
@@ -0,0 +1,169 @@
+    before(() => {
+      sandbox = sinon.sandbox.create();
+      valueStub = sandbox.stub();
+      stringStub = sandbox.stub();
+      jsonStub = sandbox.stub();
+      putStub = sandbox.stub(Etcd3.prototype, 'put', () => {
+        return {
+          value: (val) => Promise.resolve(valueStub(val))
+        };
+      });
+      getStub = sandbox.stub(Etcd3.prototype, 'get', () => {
+        return {
+          json: () => Promise.resolve(jsonStub())
+        };
+      });
+      lockStub = sandbox.stub(Etcd3.prototype, 'lock', () => {
+        return {
+          ttl: () => {
+            return {
+              acquire: () => Promise.resolve(stringStub())
+            };
+          },
+          release: () => Promise.resolve(stringStub())
+        };
+      });
+    });
+
+    afterEach(function () {
+      valueStub.reset();
+      putStub.reset();
+      getStub.reset();
+      jsonStub.reset();
+      stringStub.reset();
+    });
+    after(function () {
+      sandbox.restore();
+    });
+
+    describe('#isWriteLocked', () => {
+      it('should return false in case the resource has no lock', () => {
+        return manager.isWriteLocked('fakeResource')
+          .then(() => {
+            /* jshint expr: true */
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return true for a write lock.', () => {
+        const writeLockResp = {
+          'count': 1,
+          'operationType': 'WRITE'
+        };
+        jsonStub.onCall(0).returns(writeLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(true);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return false for a read lock.', () => {
+        const readLockResp = {
+          'count': 1,
+          'operationType': 'READ'
+        };
+        jsonStub.onCall(0).returns(readLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(false);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+      it('should return false for a no lock.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        return manager.isWriteLocked('fakeResource')
+          .then(result => {
+            /* jshint expr: true */
+            expect(result).to.eql(false);
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+    });
+
+    describe('#lock', () => {
+      it('should succeed when lock details is undefined.', () => {
```

Comment:
When do we say lock function is succeeded? How do we check lock.release() is called?

Reply:
I'll create two separate stub for release and acquire as discussed.